### PR TITLE
[PORTCLS] Remove meaningless YDEBUG

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Code cleanup.

Addendum to b77ebc4 (r54584).

Follow-up of #5818.

## Proposed changes

- Remove meaningless YDEBUG.
- Plus, 4 s/void/VOID/.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]

File: drivers/wdm/audio/backpln/portcls/purecall.cpp [hunk 1]
```diff
@@ -8,20 +8,12 @@
...
-#ifndef YDEBUG
-#define NDEBUG
-#endif
-
-#include <debug.h>
-
 extern "C" {
-  void
+  VOID
 	  __cxa_pure_virtual()
```
Comment (HBelusca) [inline drivers/wdm/audio/backpln/portcls/purecall.cpp:13]:
```suggestion
void __cxa_pure_virtual()
```
this is some CRT stuff.

File: drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp [hunk 2]
```diff
@@ -267,7 +257,7 @@ PinWaveCyclicAddEndOfStreamEvent(
     Entry->bLoopedStreaming = FALSE;
 
     // insert item
-    (void)ExInterlockedInsertTailList(&Pin->m_EventList, &EventEntry->ListEntry, &Pin->m_EventListLock);
+    (VOID)ExInterlockedInsertTailList(&Pin->m_EventList, &EventEntry->ListEntry, &Pin->m_EventListLock);
```
Comment (HBelusca) [inline drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp:260]: This one...

File: drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp [hunk 3]
```diff
@@ -308,7 +298,7 @@ PinWaveCyclicAddLoopedStreamEvent(
     DPRINT1("Added event\n");
 
     // insert item
-    (void)ExInterlockedInsertTailList(&Pin->m_EventList, &EventEntry->ListEntry, &Pin->m_EventListLock);
+    (VOID)ExInterlockedInsertTailList(&Pin->m_EventList, &EventEntry->ListEntry, &Pin->m_EventListLock);
```
Comment (HBelusca) [inline drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp:301]: ... and that one could stay as they originally were.

File: drivers/wdm/audio/backpln/portcls/interfaces.hpp [hunk 4]
```diff
@@ -924,26 +924,26 @@ typedef IPortPinDMus *PPORTPINDMUS;
...
                                                                           \
-    STDMETHODIMP_(PVOID) SystemAddress(void);                             \
+    STDMETHODIMP_(PVOID) SystemAddress(VOID);                             \
     STDMETHODIMP_(PHYSICAL_ADDRESS) PhysicalAddress(                      \
         IN  PPHYSICAL_ADDRESS PhysicalAddressConstraint OPTIONAL);        \
-    STDMETHODIMP_(PADAPTER_OBJECT) GetAdapterObject(void);                \
+    STDMETHODIMP_(PADAPTER_OBJECT) GetAdapterObject(VOID);                \
                                                                           \
-    STDMETHODIMP_(void) CopyTo(                                           \
+    STDMETHODIMP_(VOID) CopyTo(                                           \
```
Comment (HBelusca) [inline drivers/wdm/audio/backpln/portcls/interfaces.hpp:941]: Please revert all these "void" --> "VOID" changes of yours in this file, and in the corresponding implementation functions : Check official portcls.h (that defines the IDmaChannel interface amongst others), they use "void", so let's stay compatible there.

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: drivers/wdm/audio/backpln/portcls/purecall.cpp (hunk 1 above)
Comment (SergeGautherie) [reply drivers/wdm/audio/backpln/portcls/purecall.cpp:13]: Reverted.
I'll have a look separately.

File: drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp (hunk 3 above)
Comment (SergeGautherie) [reply drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp:301]: I moved the 4 remaining cases to a second commit: trivial, for consistency.

File: drivers/wdm/audio/backpln/portcls/interfaces.hpp (hunk 4 above)
Comment (SergeGautherie) [reply drivers/wdm/audio/backpln/portcls/interfaces.hpp:941]: Reverted.

File: drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp (hunk 3 above)
Comment (SergeGautherie) [reply drivers/wdm/audio/backpln/portcls/pin_wavecyclic.cpp:301]: Reverted the 4 remaining cases.

File: drivers/wdm/audio/backpln/portcls/purecall.cpp [hunk 5]
```diff
@@ -8,20 +8,12 @@
...
 
-#ifndef YDEBUG
-#define NDEBUG
-#endif
-
-#include <debug.h>
-
 extern "C" {
   void
 	  __cxa_pure_virtual()
```
Comment (HBelusca) [inline drivers/wdm/audio/backpln/portcls/purecall.cpp:13]:
```suggestion
  void __cxa_pure_virtual()
```

**Review by HBelusca — APPROVED** [review]

**Pull request merged**